### PR TITLE
Bug 1613283 - Add log rotation option for taskcluster.yaml

### PR DESCRIPTION
**Pull request by hneiva**

### CODE REVIEW

File: src/scriptworker/log.py [hunk 1]
```diff
@@ -57,6 +57,12 @@ def update_logging_config(context: Any, log_name: Optional[str] = None, file_nam
         # If we rotate the log file via logrotate.d, let's watch the file
         # so we can automatically close/reopen on move.
         handler = logging.handlers.WatchedFileHandler(path)  # type: ignore
+    elif context.config.get("log_max_bytes", 0) or context.config.get("log_max_backups", 0):
```
Comment (escapewindow) [inline src/scriptworker/log.py:60]: maybe `and` instead of `or` ?

File: src/scriptworker/log.py [hunk 2]
```diff
@@ -57,6 +57,12 @@ def update_logging_config(context: Any, log_name: Optional[str] = None, file_nam
...
+        handler = logging.handlers.RotatingFileHandler(  # type: ignore
+            filename=path,
+            maxBytes=context.config.get("log_max_bytes", 0),
+            backupCount=context.config.get("log_max_backups", 0),
```
Comment (escapewindow) [inline src/scriptworker/log.py:64]: at which point we don't need the `get`s here.

**Review by escapewindow — APPROVED** [review]
Thank you!
I do prefer if we make the changes suggested.

File: src/scriptworker/log.py (hunk 1 above)
Comment (hneiva) [reply src/scriptworker/log.py:60]: Mmmm I think we should go by `log_max_bytes` only then.
Wouldn't make sense to have unlimited max_bytes but max_backups > 0

File: src/scriptworker/log.py (hunk 2 above)
Comment (hneiva) [reply src/scriptworker/log.py:64]: Right :)

File: src/scriptworker/log.py (hunk 1 above)
Comment (escapewindow) [reply src/scriptworker/log.py]: Does this mean you want to allow for max_backups == 0 but we blow away the log every log_max_bytes?

Comment (escapewindow) [reply src/scriptworker/log.py]: If so, I'd lean towards the default of max_backups to be, say, 10 in constants.py.

**Pull request merged**